3-async-await/solution.js

```diff
@@ -20,16 +20,129 @@ Example:
 9. as extra challenge: add Promise.race() and Promise.any(), and try to get the idea of what happens
 */
 
-function solution() {
-    // YOUR SOLUTION GOES HERE
+const pricesAsync = require("./prices");
+const productsAsync = require("./products");
 
-    // You generate your id value here
+/**
+ * 
+ * @param {*} id positive number
+ * @returns returns either:
+ *                    - a string with the error message or 
+ *                    - an object as follow: { id: 1, price: 0.5, product: "salt" }
+ */
 
-    // You use Promise.all() here
+async function solution(id) {
+  // YOUR SOLUTION GOES HERE
+  const promiseAllSolution = async (id) => {
+    try {
+      let resultsAll = await Promise.all([pricesAsync(id), productsAsync(id)]);
+      return {
+        id,
+        price: resultsAll[0],
+        product: resultsAll[1],
+      };
+    } catch (error) {
+      return "Error: " + error.message;
+    }
+  };
 
-    // You use Promise.allSettled() here
+  const promiseAllSettledSolution = async (id) => {
+    const resultsAllSettled = await Promise.allSettled([
+      pricesAsync(id),
+      productsAsync(id),
+    ]);
 
-    // Log the results, or errors, here
-}
+    let errorReasons = resultsAllSettled.reduce(
+      (errorMessage, result) =>
+        errorMessage + (result.reason ? result.reason + " " : ""),
+      ""
+    );
+
+    if (errorReasons) {
+      return errorReasons;
+    }
+
+    return {
+      id,
+      price: resultsAllSettled[0].value,
+      product: resultsAllSettled[1].value,
+    };
```

Comment:
In order to be more clear in naming variables, we can use destructuring here
```suggestion
    const [priceSettled, productSettled] = resultsAllSettled
    
    return {
      id,
      price: priceSettled.value,
      product: productSettled.value,
    };
```

Reply:
Good for readability!
I've already applied this changes, and solved the gap it left after change the variable names.

I realized not all the solution methods have test, so I'll refractored the code too.